diff --git a/packages/font-glyphs/src/auto-build/transformed.ptl b/packages/font-glyphs/src/auto-build/transformed.ptl
index 39e7da72f..fc68f3269 100644
--- a/packages/font-glyphs/src/auto-build/transformed.ptl
+++ b/packages/font-glyphs/src/auto-build/transformed.ptl
@@ -1,6 +1,7 @@
 ###### Automatic builds
 $$include '../meta/macros.ptl'
 
+import [Joining] from "@iosevka/glyph/relation"
 import [linreg clamp mix fallback] from "@iosevka/util"
 import [getGrTree IsSuperscript IsSubscript AnyCvOrCherryPicking] from "@iosevka/glyph/relation"
 import [AnyLocalizedForm DotlessOrNot CvDecompose MathSansSerif Texture] from "@iosevka/glyph/relation"
@@ -443,6 +444,12 @@ glyph-block Autobuild-Transformed : begin
 			list 0x1E069 'cyrl/dze'
 			list 0x1E06A 'cyrl/dzhe'
 
+		# Numeric literal base ligation
+		createSubscripts 'numericbase' null : list
+			list null 'b'
+			list null 'o'
+			list null 'x'
+
 		createSubscripts 'denominator' DenominatorForm : list
 			list null 'zero.onum'
 			list null 'one.onum'
@@ -701,7 +708,7 @@ glyph-block Autobuild-Transformed-Texture : begin
 	glyph-block-import Recursive-Build : Fork
 	glyph-block-import Autobuild-Transformed-Shared : extendRelatedGlyphs link-relations wrapName
 
-	define [createTextureDerivatives gr extL extR _records] : begin
+	define [createTextureDerivatives gr joinDir extL extR _records] : begin
 		local { records relSets targetNameMap } : extendRelatedGlyphs gr.key _records
 		local pendingGlyphs : records.map : [record] => record.1
 
@@ -721,6 +728,7 @@ glyph-block Autobuild-Transformed-Texture : begin
 				if (glyphT && glyphO && glyphT.advanceWidth != glyphO.advanceWidth) : begin
 					gr.set glyphO targetNameMap.(glyphid)
 					create-glyph targetNameMap.(glyphid) unicode : glyph-proc
+						Joining.set currentGlyph joinDir
 						include glyphT AS_BASE ALSO_METRICS
 						set-width Width
 						if extL : begin
@@ -731,6 +739,14 @@ glyph-block Autobuild-Transformed-Texture : begin
 		# link-relations relSets
 
 	define ranges : list
+		# Adds more of ascii, allowing letters to expand into space and punctation
+		list 0x20 0x22
+		list 0x27 0x29
+		list 0x2C 0x2F
+		list 0x3A 0x3B
+		list 0x5B 0x5F
+		list 0x60 0x60
+		list 0x7B 0x7E
 		list 0x41 0x5A
 		list 0x61 0x7A
 		list 0xC0 0xFF
@@ -747,14 +763,18 @@ glyph-block Autobuild-Transformed-Texture : begin
 				if source : results.push { null source }
 		return results
 
-	define EXTENSION :  1 / 12
-	define SHRINK    : -1 / 12
-	createTextureDerivatives Texture.ExtL  EXTENSION 0         [jobs 0xF000]
-	createTextureDerivatives Texture.ExtR  0         EXTENSION [jobs 0xF100]
-	createTextureDerivatives Texture.ExtLR EXTENSION EXTENSION [jobs 0xF200]
-	createTextureDerivatives Texture.ShrL  SHRINK    0         [jobs 0xF300]
-	createTextureDerivatives Texture.ShrR  0         SHRINK    [jobs 0xF400]
-	createTextureDerivatives Texture.ShrLR SHRINK    SHRINK    [jobs 0xF500]
+	# More dramatic effect to texture extension and shrink
+	define EXTENSION     :  8 / 32
+	define DBL_EXTENSION :  4 / 32
+	define SHRINK        : -6 / 32
+	define DBL_SHRINK    : -4 / 32
+	# Adds joining classes to avoid clipping
+	createTextureDerivatives Texture.ExtL  Joining.Classes.Left  EXTENSION     0             [jobs 0xF000]
+	createTextureDerivatives Texture.ExtR  Joining.Classes.Right 0             EXTENSION     [jobs 0xF100]
+	createTextureDerivatives Texture.ExtLR Joining.Classes.Mid   DBL_EXTENSION DBL_EXTENSION [jobs 0xF200]
+	createTextureDerivatives Texture.ShrL  Joining.Classes.Left  SHRINK        0             [jobs 0xF300]
+	createTextureDerivatives Texture.ShrR  Joining.Classes.Right 0             SHRINK        [jobs 0xF400]
+	createTextureDerivatives Texture.ShrLR Joining.Classes.Mid   DBL_SHRINK    DBL_SHRINK    [jobs 0xF500]
 
 glyph-block Autobuild-Transformed-Mathematical : begin
 	glyph-block-import CommonShapes
diff --git a/packages/font-glyphs/src/letter/latin/upper-a.ptl b/packages/font-glyphs/src/letter/latin/upper-a.ptl
index 50a29c90a..aafb3eb12 100644
--- a/packages/font-glyphs/src/letter/latin/upper-a.ptl
+++ b/packages/font-glyphs/src/letter/latin/upper-a.ptl
@@ -28,7 +28,8 @@ glyph-block Letter-Latin-Upper-A : begin
 	glyph-block-export ABarPosition
 	define [ABarPosition slabKind top] : begin
 		local fBaseSlabs : maskBits slabKind : bitOr SLAB-LEFT SLAB-RIGHT
-		return : mix [if fBaseSlabs Stroke 0] top (XH / 2 / CAP)
+		# Lower the ABar by a stroke width
+		return : [mix [if fBaseSlabs Stroke 0] top (XH / 2 / CAP)] - Stroke
 
 	glyph-block-export AConfig
 	define AConfig : SuffixCfg.weave
diff --git a/packages/font-glyphs/src/letter/latin/upper-q.ptl b/packages/font-glyphs/src/letter/latin/upper-q.ptl
index 6666ea2c9..9eb695cee 100644
--- a/packages/font-glyphs/src/letter/latin/upper-q.ptl
+++ b/packages/font-glyphs/src/letter/latin/upper-q.ptl
@@ -94,9 +94,10 @@ glyph-block Letter-Latin-Upper-Q : begin
 			widths.rhs
 			flat startx HalfStroke [heading Downward]
 			curl startx 0 [heading Downward]
-			arcvh
+			# More distinctive curled tail, mirroring l and t.
+			arcvh.superness DesignParameters.tightHookSuperness
 			flat [Math.min (df.middle + HookX - 1) (startx - TailDepth - [HSwToV Stroke])] (Stroke + TailDepth)
-			curl (df.middle + HookX) (Stroke + TailDepth)
+			curl (df.middle + HookX + Stroke) (Stroke + TailDepth)
 
 	define [QCrossingCurlyTail df top sw] : union
 		VBar.m df.middle HalfStroke [mix Descender HalfStroke 1.75] sw
diff --git a/packages/font-glyphs/src/meta/aesthetics.ptl b/packages/font-glyphs/src/meta/aesthetics.ptl
index b546cf30d..856a4a9c3 100644
--- a/packages/font-glyphs/src/meta/aesthetics.ptl
+++ b/packages/font-glyphs/src/meta/aesthetics.ptl
@@ -383,7 +383,8 @@ export : define DesignParameters : object
 	geometric_small_x          (4    / 7)
 	geometric_medium_small_x   (4.75 / 7)
 	geometric_medium_x         (5.5  / 7)
-	arrow_size                 0.45
+	# TODO: can remove this if its no longer used
+	arrow_size                 0.585
 	# ()
 	parenOutside               0.2
 	parenOutsideSwAdj          0.2
diff --git a/packages/font-glyphs/src/number/0.ptl b/packages/font-glyphs/src/number/0.ptl
index b7218ad6a..b52a4ca47 100644
--- a/packages/font-glyphs/src/number/0.ptl
+++ b/packages/font-glyphs/src/number/0.ptl
@@ -151,7 +151,8 @@ glyph-block Digits-Zero : begin
 			MaskBelow (top * (1/2 + 1/16))
 
 	define [ZeroDotShape top] : begin
-		local halfDotWidth : Math.min DotRadius (CircleInnerWidth / 4)
+		# Dot to fill 1/3 of inner width
+		local halfDotWidth : Math.min DotRadius (CircleInnerWidth / 6)
 		return : OShapeOutline.NoOvershoot
 			top / 2 + DotRadius
 			top / 2 - DotRadius
@@ -162,7 +163,7 @@ glyph-block Digits-Zero : begin
 			ArchDepthBOf halfDotWidth (halfDotWidth * 2)
 
 	define [ZeroLongDotShape top] : begin
-		local halfDotWidth : Math.min HalfStroke (CircleInnerWidth / 4)
+		local halfDotWidth : Math.min HalfStroke (CircleInnerWidth / 6)
 		local halfDotHeight : Math.max DotRadius : Math.min ((top - Stroke * 2) / 4) ((top - Stroke * 2 - (CircleInnerWidth - halfDotWidth * 2)) / 2)
 		return : OShapeOutline.NoOvershoot
 			top / 2 + halfDotHeight
@@ -174,7 +175,7 @@ glyph-block Digits-Zero : begin
 			ArchDepthBOf halfDotWidth (halfDotWidth * 2)
 
 	define [ZeroOvalDotShape top] : begin
-		local halfDotWidth : Math.min DotRadius (CircleInnerWidth / 4)
+		local halfDotWidth : Math.min DotRadius (CircleInnerWidth / 6)
 		return : OvalShapeT spiro-outline
 			top / 2 + DotRadius - O
 			top / 2 - DotRadius + O
@@ -183,7 +184,7 @@ glyph-block Digits-Zero : begin
 			begin 0
 
 	define [ZeroOvalLongDotShape top] : begin
-		local halfDotWidth : Math.min HalfStroke (CircleInnerWidth / 4)
+		local halfDotWidth : Math.min HalfStroke (CircleInnerWidth / 6)
 		local halfDotHeight : Math.max DotRadius : Math.min ((top - Stroke * 2) / 4) ((top - Stroke * 2 - (CircleInnerWidth - halfDotWidth * 2)) / 2)
 		return : OvalShapeT spiro-outline
 			top / 2 + halfDotHeight - O
diff --git a/packages/font-glyphs/src/number/6.ptl b/packages/font-glyphs/src/number/6.ptl
index 6f6947cbb..36773a69e 100644
--- a/packages/font-glyphs/src/number/6.ptl
+++ b/packages/font-glyphs/src/number/6.ptl
@@ -11,7 +11,8 @@ glyph-block Digits-Six : begin
 
 	glyph-block-export OpenContourSixShape
 	define [OpenContourSixShape charTop] : glyph-proc
-		local oTop : charTop * 0.55 + Stroke / 2
+		# Exagerate o in 6 and 9 to be closer to XH
+		local oTop : charTop * 0.6 + Stroke / 2
 		local fine   : Stroke * CThinB
 		local coFine : Math.max [AdviceStroke 3.5] (Stroke - fine / 2)
 		include : dispiro
@@ -51,7 +52,8 @@ glyph-block Digits-Six : begin
 
 	glyph-block-export StraightBarSixShape
 	define [StraightBarSixShape] : with-params [charTop [sw Stroke] [bbd 0]] : glyph-proc
-		local oTop : charTop * 0.55 + sw / 2
+		# Exagerate o in 6 and 9 to be closer to XH
+		local oTop : charTop * 0.6074 + sw / 2
 		local fine   : sw * CThin
 		local coFine : Math.max (sw * CThin) (sw - fine / 2)
 
diff --git a/packages/font-glyphs/src/symbol/arrow.ptl b/packages/font-glyphs/src/symbol/arrow.ptl
index 557cd7cd8..765429fdb 100644
--- a/packages/font-glyphs/src/symbol/arrow.ptl
+++ b/packages/font-glyphs/src/symbol/arrow.ptl
@@ -8,35 +8,56 @@ glyph-block Symbol-Arrow : for-width-kinds WideWidth1
 	glyph-block-import CommonShapes
 	glyph-block-import Common-Derivatives
 	glyph-block-import Symbol-Geometric-Shared : UnicodeWeightGrade
+	glyph-block-import Symbol-Math-Relation-Common : EqualHalfSpace IdentHalfSpace
 
-	define arrowHeight : para.arrowHeight * (Width - SB / 2) * [Math.pow (MosaicWidth / MosaicUnitWidth) (1 / 3)]
-	define arrowWidth  : Math.min (MosaicWidth - SB / 2) arrowHeight
-	define arrowTop : [mix OperBot OperTop (1 / 2)] + arrowHeight / 2
-	define arrowBot : [mix OperBot OperTop (1 / 2)] - arrowHeight / 2
+	# Arrow width as the width of two operators, marginless in fixed width
+	define arrowWidth : MosaicWidth - SB * 4 * (MosaicWidthScalar - 1)
+	define arrowHeight : Width * 2 - SB * 4
+
+	# Arrow vertically baseline aligned
+	define arrowTop  arrowHeight
+	define arrowBot  0
 	define arrowMidX : MosaicWidth / 2
 	define arrowSB : (MosaicWidth - arrowWidth) / 2
 	define arrowRSB : MosaicWidth - arrowSB
 	define o : O * 2
 
-	define arrowHeadSize : (Width - SB) * DesignParameters.arrow_size * [Math.pow (arrowWidth / Width) 0.5] - [AdviceStroke 5] / 4
 	define arrowSw       OperatorStroke
+	# Arrow head relative to symbol midline and baseline
+	define arrowHeadSize : (SymbolMid - arrowSw * 0.5 * Math.SQRT1_2) * [Math.pow (MosaicWidthScalar / 2) 0.5]
+	define vertArrowHeadSize : Math.min arrowWidth (SymbolMid - arrowSw * 0.5 * Math.SQRT1_2)
 	define halfArrowSw : arrowSw / 2
 	define bendArrowHeadSize : mix arrowSw arrowHeadSize 0.75
 	define fine : AdviceStroke 3.5
 	define terminal : AdviceStroke 5
 	define halfTerminal : terminal / 2
-	define doubleArrowStemWidth : arrowHeadSize * 1.1
+	# Match ligature arrows
+	define doubleArrowStemWidth : arrowSw + EqualHalfSpace * 2
+	define tripleArrowStemWidth : arrowSw + IdentHalfSpace * 2
 
 	define kLongArrowSB : [DivFrame MosaicWidthScalar 4].leftSB / [DivFrame MosaicWidthScalar 2].leftSB
 
 	# Diagonal arrow params
-	define diagPropX : 1 / [Math.pow 2 (1 / 4)]
-	define diagPropY : 1 / [Math.pow 2 (1 / 4)]
+	define diagPropX : 1 / [Math.pow 2 (2 / 5)]
+	define diagPropY : 1 / [Math.pow 2 (2 / 5)]
 	define arrowDiagRSB : mix arrowMidX arrowRSB diagPropX
 	define arrowDiagSB  : mix arrowMidX arrowSB  diagPropX
 	define arrowDiagBot : mix arrowMidX arrowBot diagPropY
 	define arrowDiagTop : mix arrowMidX arrowTop diagPropY
 
+	# Double arrow params
+	define arrowDoubleHeight : arrowHeight * 1.25
+	define arrowDoubleWidth : Math.min (MosaicWidth - SB) arrowDoubleHeight
+	define arrowDoubleTop : [mix OperBot OperTop (1 / 2)] + arrowDoubleHeight / 2
+	define arrowDoubleBot : [mix OperBot OperTop (1 / 2)] - arrowDoubleHeight / 2
+	define arrowDoubleMidX : MosaicWidth / 2
+	define arrowDoubleSB : (MosaicWidth - arrowDoubleWidth) / 2
+	define arrowDoubleRSB : MosaicWidth - arrowDoubleSB
+	define arrowDoubleDiagRSB : mix arrowMidX arrowDoubleRSB diagPropX
+	define arrowDoubleDiagSB  : mix arrowMidX arrowDoubleSB  diagPropX
+	define arrowDoubleDiagBot : mix arrowMidX arrowDoubleBot diagPropY
+	define arrowDoubleDiagTop : mix arrowMidX arrowDoubleTop diagPropY
+
 	define [ArrowBar x1 y1 x2 y2 halfSw w _offset] : begin
 		return : PointingTo x1 y1 x2 y2 : lambda [mag] : begin
 			local p : (mag - [fallback _offset : o + halfSw * [fallback w 1.1]]) / mag
@@ -90,23 +111,32 @@ glyph-block Symbol-Arrow : for-width-kinds WideWidth1
 				flat (mag - [Math.abs s] / 2) 0 [widths.center (halfSw * 2)]
 				curl [mix mag 0 p] 0
 
-	define [ArrowHead x1 y1 x2 y2 _size] : union
-		LHSHalfArrowHead x1 y1 x2 y2 0 _size
-		RHSHalfArrowHead x1 y1 x2 y2 0 _size
+	# Squared arrow head
+	define [ArrowHead x1 y1 x2 y2 _size] : begin
+		local size : fallback _size arrowHeadSize
+		return : union
+			new-glyph : PointingTo x1 y1 x2 y2 : lambda [mag] : dispiro
+				widths.lhs arrowSw
+				flat (o) 0
+				curl (o + size) (-size)
+			new-glyph : PointingTo x1 y1 x2 y2 : lambda [mag] : dispiro
+				widths.rhs arrowSw
+				flat (o) 0
+				curl (o + size) size
 
 	define [LHSHalfArrowHead x1 y1 x2 y2 width _size] : begin
 		local size : fallback _size arrowHeadSize
 		return : new-glyph : PointingTo x1 y1 x2 y2 : lambda [mag] : dispiro
-			widths.lhs fine
+			widths.lhs arrowSw
 			flat (o) (width / 2) [heading Downward]
-			curl (o + size) (-size) [widths.lhs terminal]
+			curl (o + size) (-size)
 
 	define [RHSHalfArrowHead x1 y1 x2 y2 width _size] : begin
 		local size : fallback _size arrowHeadSize
 		return : new-glyph : PointingTo x1 y1 x2 y2 : lambda [mag] : dispiro
-			widths.rhs fine
+			widths.rhs arrowSw
 			flat (o) (-width / 2) [heading Upward]
-			curl (o + size) size [widths.rhs terminal]
+			curl (o + size) size
 
 	define [SsArrowHead x1 y1 x2 y2 _size] : union
 		LHSHalfSsArrowHead x1 y1 x2 y2 0 _size
@@ -524,24 +554,24 @@ glyph-block Symbol-Arrow : for-width-kinds WideWidth1
 
 	do "Straight Arrow"
 		MkArrow ArrowShape [MangleName 'arrowLeft']       [MangleUnicode 0x2190] arrowRSB SymbolMid arrowSB SymbolMid
-		MkArrow ArrowShape [MangleName 'arrowUp']         [MangleUnicode 0x2191] arrowMidX arrowBot arrowMidX arrowTop
+		MkArrow ArrowShape [MangleName 'arrowUp']         [MangleUnicode 0x2191] arrowMidX arrowBot arrowMidX arrowTop nothing vertArrowHeadSize
 		MkArrow ArrowShape [MangleName 'arrowRight']      [MangleUnicode 0x2192] arrowSB SymbolMid arrowRSB SymbolMid
-		MkArrow ArrowShape [MangleName 'arrowDown']       [MangleUnicode 0x2193] arrowMidX arrowTop arrowMidX arrowBot
+		MkArrow ArrowShape [MangleName 'arrowDown']       [MangleUnicode 0x2193] arrowMidX arrowTop arrowMidX arrowBot nothing vertArrowHeadSize
 		MkArrow ArrowShape [MangleName 'arrowUpLeft']     [MangleUnicode 0x2196] arrowDiagRSB arrowDiagBot arrowDiagSB arrowDiagTop
 		MkArrow ArrowShape [MangleName 'arrowUpRight']    [MangleUnicode 0x2197] arrowDiagSB arrowDiagBot arrowDiagRSB arrowDiagTop
 		MkArrow ArrowShape [MangleName 'arrowDownRight']  [MangleUnicode 0x2198] arrowDiagSB arrowDiagTop arrowDiagRSB arrowDiagBot
 		MkArrow ArrowShape [MangleName 'arrowDownLeft']   [MangleUnicode 0x2199] arrowDiagRSB arrowDiagTop arrowDiagSB arrowDiagBot
 
 		MkArrow ThinArrowShape [MangleName 'thinArrowLeft']  null arrowRSB SymbolMid arrowSB SymbolMid
-		MkArrow ThinArrowShape [MangleName 'thinArrowUp']    null arrowMidX arrowBot arrowMidX arrowTop
+		MkArrow ThinArrowShape [MangleName 'thinArrowUp']    null arrowMidX arrowBot arrowMidX arrowTop nothing vertArrowHeadSize
 		MkArrow ThinArrowShape [MangleName 'thinArrowRight'] null arrowSB SymbolMid arrowRSB SymbolMid
-		MkArrow ThinArrowShape [MangleName 'thinArrowDown']  null arrowMidX arrowTop arrowMidX arrowBot
+		MkArrow ThinArrowShape [MangleName 'thinArrowDown']  null arrowMidX arrowTop arrowMidX arrowBot nothing vertArrowHeadSize
 
 		MkArrow TrigArrowShape [MangleName 'uni279D']             [MangleUnicode 0x279D] arrowSB SymbolMid arrowRSB SymbolMid
 		MkArrow TrigArrowShape [MangleName 'trigArrowLeft']       [MangleUnicode 0x2B60] arrowRSB SymbolMid arrowSB SymbolMid
-		MkArrow TrigArrowShape [MangleName 'trigArrowUp']         [MangleUnicode 0x2B61] arrowMidX arrowBot arrowMidX arrowTop
+		MkArrow TrigArrowShape [MangleName 'trigArrowUp']         [MangleUnicode 0x2B61] arrowMidX arrowBot arrowMidX arrowTop nothing vertArrowHeadSize
 		MkArrow TrigArrowShape [MangleName 'trigArrowRight']      [MangleUnicode 0x2B62] arrowSB SymbolMid arrowRSB SymbolMid
-		MkArrow TrigArrowShape [MangleName 'trigArrowDown']       [MangleUnicode 0x2B63] arrowMidX arrowTop arrowMidX arrowBot
+		MkArrow TrigArrowShape [MangleName 'trigArrowDown']       [MangleUnicode 0x2B63] arrowMidX arrowTop arrowMidX arrowBot nothing vertArrowHeadSize
 		MkArrow TrigArrowShape [MangleName 'trigArrowUpLeft']     [MangleUnicode 0x2B66] arrowDiagRSB arrowDiagBot arrowDiagSB arrowDiagTop
 		MkArrow TrigArrowShape [MangleName 'trigArrowUpRight']    [MangleUnicode 0x2B67] arrowDiagSB arrowDiagBot arrowDiagRSB arrowDiagTop
 		MkArrow TrigArrowShape [MangleName 'trigArrowDownRight']  [MangleUnicode 0x2B68] arrowDiagSB arrowDiagTop arrowDiagRSB arrowDiagBot
@@ -567,9 +597,9 @@ glyph-block Symbol-Arrow : for-width-kinds WideWidth1
 
 	do "Dash Straight Arrow"
 		MkArrow DashArrowShape [MangleName 'dashArrowLeft']  [MangleUnicode 0x21E0] arrowRSB SymbolMid arrowSB SymbolMid
-		MkArrow DashArrowShape [MangleName 'dashArrowUp']    [MangleUnicode 0x21E1] arrowMidX arrowBot arrowMidX arrowTop
+		MkArrow DashArrowShape [MangleName 'dashArrowUp']    [MangleUnicode 0x21E1] arrowMidX arrowBot arrowMidX arrowTop nothing vertArrowHeadSize
 		MkArrow DashArrowShape [MangleName 'dashArrowRight'] [MangleUnicode 0x21E2] arrowSB SymbolMid arrowRSB SymbolMid
-		MkArrow DashArrowShape [MangleName 'dashArrowDown']  [MangleUnicode 0x21E3] arrowMidX arrowTop arrowMidX arrowBot
+		MkArrow DashArrowShape [MangleName 'dashArrowDown']  [MangleUnicode 0x21E3] arrowMidX arrowTop arrowMidX arrowBot nothing vertArrowHeadSize
 
 		MkArrow TrigDashArrowShape [MangleName 'trigDashArrowLeft']  [MangleUnicode 0x2B6A] arrowRSB SymbolMid arrowSB SymbolMid
 		MkArrow TrigDashArrowShape [MangleName 'trigDashArrowUp']    [MangleUnicode 0x2B6B] arrowMidX arrowBot arrowMidX arrowTop
@@ -731,7 +761,8 @@ glyph-block Symbol-Arrow : for-width-kinds WideWidth1
 		DoubleArrow2 [HalfArrowShape LHSHalfArrowHead] [HalfArrowShape RHSHalfArrowHead] [MangleName 'arrowUpHLDownHR']    [MangleUnicode 0x2951] arrowMidX arrowTop arrowMidX arrowBot
 
 
-	define waveArrowAmplitude : (Width - SB) * DesignParameters.arrow_size * (0.7 + 0.4 * MosaicWidthScalar)
+    # Match wave ligations
+	define waveArrowAmplitude : (EqualHalfSpace + OperatorStroke) / 0.6 # (Width - SB) * DesignParameters.arrow_size * (0.7 + 0.4 * MosaicWidthScalar)
 	do "Wave arrows"
 		glyph-block-import Shared-Symbol-Shapes : CreateWaveShape
 		define WaveSw : AdviceStroke (5.5 - MosaicWidthScalar)
@@ -931,14 +962,14 @@ glyph-block Symbol-Arrow : for-width-kinds WideWidth1
 			MkArrow [DoubleArrowShape dasw] 'longDblArrowRight' 0x27F9 l SymbolMid r SymbolMid
 
 	do "Triple-line Arrows"
-		define dasw doubleArrowStemWidth
+		define dasw tripleArrowStemWidth
 		MkArrow [TripleArrowShape dasw] [MangleName 'tripleArrowLeft']    [MangleUnicode 0x21DA] arrowRSB SymbolMid arrowSB SymbolMid
 		MkArrow [TripleArrowShape dasw] [MangleName 'tripleArrowRight']   [MangleUnicode 0x21DB] arrowSB SymbolMid arrowRSB SymbolMid
 		MkArrow [TripleArrowShape dasw] [MangleName 'tripleArrowUp']      [MangleUnicode 0x290A] arrowMidX arrowBot arrowMidX arrowTop
 		MkArrow [TripleArrowShape dasw] [MangleName 'tripleArrowDown']    [MangleUnicode 0x290B] arrowMidX arrowTop arrowMidX arrowBot
 
 	do "Quadruple-line Arrows"
-		define dasw doubleArrowStemWidth
+		define dasw tripleArrowStemWidth
 		MkArrow [QuadrupleArrowShape dasw] [MangleName 'quadrupleArrowUp']      [MangleUnicode 0x27F0] arrowMidX arrowBot arrowMidX arrowTop
 		MkArrow [QuadrupleArrowShape dasw] [MangleName 'quadrupleArrowDown']    [MangleUnicode 0x27F1] arrowMidX arrowTop arrowMidX arrowBot
 		MkArrow [QuadrupleArrowShape dasw] [MangleName 'quadrupleArrowLeft']    [MangleUnicode 0x2B45] arrowRSB SymbolMid arrowSB SymbolMid
@@ -1098,17 +1129,17 @@ glyph-block Symbol-Arrow : for-width-kinds WideWidth1
 				VBar.m l (SymbolMid - arrowHeadSize) (SymbolMid + arrowHeadSize) fine
 
 	do "Double-headed Arrows"
-		DoubleArrow ArrowShape [MangleName 'arrowLeftRight'] [MangleUnicode 0x2194] arrowSB SymbolMid arrowRSB SymbolMid
-		DoubleArrow ThinArrowShape [MangleName 'thinArrowLeftRight'] null arrowSB SymbolMid arrowRSB SymbolMid
-		DoubleArrow ArrowShape [MangleName 'arrowUpDown'] [MangleUnicode 0x2195] arrowMidX arrowTop arrowMidX arrowBot
-		DoubleArrow ArrowShape [MangleName 'arrowUpDown.bottomSpaceLeft'] nothing arrowMidX arrowTop arrowMidX (arrowBot + arrowSw * 1.5)
-		DoubleArrow ArrowShape [MangleName 'arrowUpLeftAndDownRight'] [MangleUnicode 0x2921] arrowDiagSB arrowDiagTop arrowDiagRSB arrowDiagBot
-		DoubleArrow ArrowShape [MangleName 'arrowUpRightAndDownLeft'] [MangleUnicode 0x2922] arrowDiagRSB arrowDiagTop arrowDiagSB arrowDiagBot
+		DoubleArrow ArrowShape [MangleName 'arrowLeftRight'] [MangleUnicode 0x2194] arrowDoubleSB SymbolMid arrowDoubleRSB SymbolMid
+		DoubleArrow ThinArrowShape [MangleName 'thinArrowLeftRight'] null arrowDoubleSB SymbolMid arrowDoubleRSB SymbolMid
+		DoubleArrow ArrowShape [MangleName 'arrowUpDown'] [MangleUnicode 0x2195] arrowMidX arrowDoubleTop arrowMidX arrowDoubleBot
+		DoubleArrow ArrowShape [MangleName 'arrowUpDown.bottomSpaceLeft'] nothing arrowMidX arrowDoubleTop arrowMidX (arrowDoubleBot + arrowSw * 1.5)
+		DoubleArrow ArrowShape [MangleName 'arrowUpLeftAndDownRight'] [MangleUnicode 0x2921] arrowDoubleDiagSB arrowDoubleDiagTop arrowDoubleDiagRSB arrowDoubleDiagBot
+		DoubleArrow ArrowShape [MangleName 'arrowUpRightAndDownLeft'] [MangleUnicode 0x2922] arrowDoubleDiagRSB arrowDoubleDiagTop arrowDoubleDiagSB arrowDoubleDiagBot
 
-		DoubleArrow TrigArrowShape [MangleName 'trigArrowLeftRight'] [MangleUnicode 0x2B64] arrowSB SymbolMid arrowRSB SymbolMid
-		DoubleArrow TrigArrowShape [MangleName 'trigArrowUpDown'] [MangleUnicode 0x2B65] arrowMidX arrowTop arrowMidX arrowBot
+		DoubleArrow TrigArrowShape [MangleName 'trigArrowLeftRight'] [MangleUnicode 0x2B64] arrowDoubleSB SymbolMid arrowDoubleRSB SymbolMid
+		DoubleArrow TrigArrowShape [MangleName 'trigArrowUpDown'] [MangleUnicode 0x2B65] arrowDoubleMidX arrowDoubleTop arrowMidX arrowDoubleBot
 
-		DoubleArrow OpenArrowShape [MangleName 'openHeadArrowLeftRight'] [MangleUnicode 0x21FF] arrowSB SymbolMid arrowRSB SymbolMid
+		DoubleArrow OpenArrowShape [MangleName 'openHeadArrowLeftRight'] [MangleUnicode 0x21FF] arrowDoubleSB SymbolMid arrowDoubleRSB SymbolMid
 
 		create-glyph [MangleName 'arrowUpDownBar'] [MangleUnicode 0x21A8] : composite-proc
 			refer-glyph : MangleName 'arrowUpDown.bottomSpaceLeft'
@@ -1121,8 +1152,8 @@ glyph-block Symbol-Arrow : for-width-kinds WideWidth1
 
 	do "Double-line Double-headed Arrows"
 		define dasw doubleArrowStemWidth
-		DoubleArrow [DoubleArrowShape dasw] [MangleName 'dblArrowLeftRight'] [MangleUnicode 0x21D4] arrowSB SymbolMid arrowRSB SymbolMid
-		DoubleArrow [DoubleArrowShape dasw] [MangleName 'dblArrowUpDown'] [MangleUnicode 0x21D5] arrowMidX arrowTop arrowMidX arrowBot
+		DoubleArrow [DoubleArrowShape dasw] [MangleName 'dblArrowLeftRight'] [MangleUnicode 0x21D4] arrowDoubleSB SymbolMid arrowDoubleRSB SymbolMid
+		DoubleArrow [DoubleArrowShape dasw] [MangleName 'dblArrowUpDown'] [MangleUnicode 0x21D5] arrowDoubleMidX arrowDoubleTop arrowDoubleMidX arrowDoubleBot
 
 		if (MosaicWidthScalar > 1 && !para.forceMonospace) : do
 			define l : mix 0 arrowSB kLongArrowSB
@@ -1130,10 +1161,10 @@ glyph-block Symbol-Arrow : for-width-kinds WideWidth1
 			DoubleArrow [DoubleArrowShape dasw] 'longDblArrowLeftRight' 0x27FA l SymbolMid r SymbolMid
 
 	do "Same-side Double-headed Arrows"
-		MkArrow dhArrowShape [MangleName 'dhArrowLeft'] [MangleUnicode 0x219E] arrowRSB SymbolMid arrowSB SymbolMid
-		MkArrow dhArrowShape [MangleName 'dhArrowRight'] [MangleUnicode 0x21A0] arrowSB SymbolMid arrowRSB SymbolMid
-		MkArrow dhArrowShape [MangleName 'dhArrowUp'] [MangleUnicode 0x219F] arrowMidX arrowBot arrowMidX arrowTop
-		MkArrow dhArrowShape [MangleName 'dhArrowDown'] [MangleUnicode 0x21A1] arrowMidX arrowTop arrowMidX arrowBot
+		MkArrow dhArrowShape [MangleName 'dhArrowLeft'] [MangleUnicode 0x219E] arrowRSB SymbolMid arrowSB SymbolMid nothing bendArrowHeadSize
+		MkArrow dhArrowShape [MangleName 'dhArrowRight'] [MangleUnicode 0x21A0] arrowSB SymbolMid arrowRSB SymbolMid nothing bendArrowHeadSize
+		MkArrow dhArrowShape [MangleName 'dhArrowUp'] [MangleUnicode 0x219F] arrowMidX arrowBot arrowMidX arrowTop nothing bendArrowHeadSize
+		MkArrow dhArrowShape [MangleName 'dhArrowDown'] [MangleUnicode 0x21A1] arrowMidX arrowTop arrowMidX arrowBot nothing bendArrowHeadSize
 
 	do "Head-and-Tail Arrows"
 		MkArrow htArrowShape [MangleName 'htArrowLeft'] [MangleUnicode 0x21A2] arrowRSB SymbolMid arrowSB SymbolMid
diff --git a/packages/font-glyphs/src/symbol/ligation.ptl b/packages/font-glyphs/src/symbol/ligation.ptl
index 54067188f..09a148d4a 100644
--- a/packages/font-glyphs/src/symbol/ligation.ptl
+++ b/packages/font-glyphs/src/symbol/ligation.ptl
@@ -10,6 +10,7 @@ glyph-module
 glyph-block Symbol-Ligation : begin
 	glyph-block-import CommonShapes
 	glyph-block-import Common-Derivatives
+	glyph-block-import Mark-Above : aboveMarkTop
 	glyph-block-import Symbol-Math-Relation-Common : LessSlope EqualHalfSpace EqualHoleWidth
 	glyph-block-import Symbol-Math-Relation-Equal : EqualShape EqualHole IdentShape IdentHole
 	glyph-block-import Symbol-Math-Arith : PlusShape
@@ -22,17 +23,30 @@ glyph-block Symbol-Ligation : begin
 		if (rk !== 'f') : set k : k + Joining.Classes.Right
 		Joining.set currentGlyph k
 
-	define RodBarHGap : Math.max (Width / 8) [AdviceStroke 6]
+	# Change to 0 to remove the gap in Logic ligations
+	define RodBarHGap 0
 	define RodBarHOffset : 0.5 * Width - [HSwToV HalfStroke] - RodBarHGap
 
 	# Arrow parameters
-	define arrowheadSlope : 1 / LessSlope
-
-	local arrowExtend : clamp 0 Width (RightSB - 0.5 * OperatorStroke * arrowheadSlope)
-	local arrowJ      : clamp 0 Width (SB + 0.5 * OperatorStroke * arrowheadSlope)
-
-	local dblExtend : clamp 0 Width (RightSB - EqualHalfSpace * arrowheadSlope)
-	local dblArrowJ : clamp 0 Width (SB + EqualHalfSpace * arrowheadSlope)
+	define arrowheadSlope (1 / 2)
+
+	# Mirror unicode arrows
+	# Ideally this would be imported from "Symbol-Arrow", or a better fix would be
+	# to move these to a "common" glyph module. But for convenience they are just copied.
+	local arrowSw         OperatorStroke
+	local arrowWidth    : Width * 2 - SB * 4
+	local arrowSB       : Width - arrowWidth / 2
+	local arrowRSB      : Width - arrowSB
+	local arrowHeadSize : SymbolMid - arrowSw * 0.5 * Math.SQRT1_2
+	local o : O * 2
+
+	local arrowShift  : OperatorStroke * arrowheadSlope - o
+	local arrowExtend : clamp 0 Width (arrowRSB - arrowShift)
+	local arrowJ      : clamp 0 Width (arrowSB + arrowShift)
+
+	local dblShift  : (EqualHalfSpace + OperatorStroke) * arrowheadSlope - o
+	local dblExtend : clamp 0 Width (arrowRSB - dblShift)
+	local dblArrowJ : clamp 0 Width (arrowSB + dblShift)
 
 	define [HyphenNotch x] : union
 		VBar.m x (SymbolMid + OperatorStroke / 3) (SymbolMid + OperatorStroke) EqualHoleWidth
@@ -43,18 +57,15 @@ glyph-block Symbol-Ligation : begin
 		define [ArrowBarConfig extend kern join doubleBar] : begin
 			define cleft O
 			define cright (Width - O)
-			define arrowr : if doubleBar
-				Width + SB + [HSwToV OperatorStroke]
-				Width + extend
-			define arrowl : if doubleBar
-				-Width + RightSB - [HSwToV OperatorStroke]
-				0 - extend
-			define zleft : cleft - Width + RightSB
-			define zright : cright + SB
+			# Ensure double and single arrows use the same bar measurements
+			define arrowr : Width + extend
+			define arrowl : 0 - extend
+			define zleft : cleft - Width + arrowRSB
+			define zright : cright + arrowSB
 			define lk : object
-				x    (SB - Width)
+				x    (arrowSB - Width)
 				y    (-join - Width)
-				f    SB
+				f    arrowSB
 				p    (arrowl + Width)
 				l    arrowl
 				l1 : arrowl - kern
@@ -63,12 +74,13 @@ glyph-block Symbol-Ligation : begin
 				j    (-join)
 				j1   (-join - kern)
 				z    zleft
-				lx : mix arrowl 0 0.5
+				# Replaced with a variant that gives the arrow head more space
+				lx   (cleft + (arrowRSB - Width))
 
 			define rk : object
-				x    (RightSB + Width)
+				x    (arrowRSB + Width)
 				y    (2 * Width + join)
-				f    RightSB
+				f    arrowRSB
 				p    (arrowr - Width)
 				r    arrowr
 				r1 : arrowr + kern
@@ -102,10 +114,10 @@ glyph-block Symbol-Ligation : begin
 						include : SetJoiningKind lS rS
 						include : difference
 							Shape left right
-							if (fNotch && left < (SB - Width))       [Notch (-Width)]    [no-shape]
-							if (fNotch && left < SB)                 [Notch 0]           [no-shape]
-							if (fNotch && right > RightSB)           [Notch Width]       [no-shape]
-							if (fNotch && right > (RightSB + Width)) [Notch (2 * Width)] [no-shape]
+							if (fNotch && left < (arrowSB - Width))   [Notch (-Width)]    [no-shape]
+							if (fNotch && left < arrowSB)             [Notch 0]           [no-shape]
+							if (fNotch && right > arrowRSB)           [Notch Width]       [no-shape]
+							if (fNotch && right > (arrowRSB + Width)) [Notch (2 * Width)] [no-shape]
 
 				select-variant "\(prefix).lig.\(lS)\(rS)"
 					follow -- "lig-\(prefix)"
@@ -115,12 +127,14 @@ glyph-block Symbol-Ligation : begin
 
 	do "Waves"
 		glyph-block-import Shared-Symbol-Shapes : CreateWaveShape
-		define WaveShape : CreateWaveShape (EqualHalfSpace + 0.5 * OperatorStroke) OperatorStroke (-1 / 4)
+		# Make wave about the same height as a tilde, remove phase shift
+		define WaveShape : CreateWaveShape (EqualHalfSpace + OperatorStroke) OperatorStroke 0
 
 		define LR  (2  / 32)
 		define LR1 (9  / 32)
-		define Z   (1  / 4)
-		define F   (-1 / 4)
+		define Z   (1  / 8)
+		define F   (-1 / 8)
+		define F1  (-1 / 8)
 		define C   0
 		define J   (-1 / 8)
 
@@ -128,6 +142,7 @@ glyph-block Symbol-Ligation : begin
 			l    { LR   (-arrowExtend)                           false }
 			l1   { LR1  (-arrowExtend - arrowheadsKern)          false }
 			f    { F    nothing                                  false }
+			f1   { F1   nothing                                  false }
 			z    { Z    nothing                                  false }
 			c    { C    nothing                                  false }
 			m    { J    (0 - RodBarHOffset)                       true }
@@ -138,12 +153,19 @@ glyph-block Symbol-Ligation : begin
 			r    { LR   (Width + arrowExtend)                    false }
 			r1   { LR1  (Width + arrowExtend + arrowheadsKern)   false }
 			f    { F    nothing                                  false }
+			f1   { F1   nothing                                  false }
 			z    { Z    nothing                                  false }
 			c    { C    nothing                                  false }
 			m    { J    (Width + RodBarHOffset)                   true }
 			j    { J    (Width + arrowJ)                          true }
 			j1   { J    (Width + arrowJ + arrowheadsKern)         true }
 
+        # Along with the above, applies frequency shaping to short waves
+        # this allows the wave ends to better match the tilde shape it replaces
+		define shapeFreqs : object
+			f1c  (-1.2)
+			cf1  (1.2)
+
 		foreach [lS : items-of : Object.keys lk] : foreach [rS : items-of : Object.keys rk] : create-glyph"wave.lig.\(lS)\(rS)" : glyph-proc
 			include : SetJoiningKind lS rS
 			local {tL jL djL} lk.(lS)
@@ -155,15 +177,35 @@ glyph-block Symbol-Ligation : begin
 				xfJoin -- jR
 				diagJoinS -- djL
 				diagJoinF -- djR
+				shapeFreq -- [fallback shapeFreqs.("\(lS)\(rS)") 0]
 
 
 	do "Arrowheads"
 		glyph-block-import Symbol-Math-Relation-Common : dH
 		glyph-block-import Symbol-Math-Relation-Inequality : LigationLessShape LigationGreaterShape
 
+		define [ArrowHead x1 y1 x2 y2 _size] : union
+			LHSHalfArrowHead x1 y1 x2 y2 0 _size
+			RHSHalfArrowHead x1 y1 x2 y2 0 _size
+
+		define [LHSHalfArrowHead x1 y1 x2 y2 width _size] : begin
+			local size : fallback _size arrowHeadSize
+			return : new-glyph : PointingTo x1 y1 x2 y2 : lambda [mag] : dispiro
+				widths.lhs arrowSw
+				flat (o) (width / 2)
+				curl (o + size) (-size)
+
+		define [RHSHalfArrowHead x1 y1 x2 y2 width _size] : begin
+			local size : fallback _size arrowHeadSize
+			return : new-glyph : PointingTo x1 y1 x2 y2 : lambda [mag] : dispiro
+				widths.rhs arrowSw
+				flat (o) (-width / 2)
+				curl (o + size) (size)
+
 		create-glyph 'less.lig.dblhead' : glyph-proc
 			Joining.set currentGlyph Joining.Classes.Right
-			include : LigationLessShape (SymbolMid + dH) (SymbolMid - dH) SB RightSB OperatorStroke OperatorStroke (2 * EqualHalfSpace)
+			# Replaced with an arrow head
+			include : ArrowHead arrowRSB SymbolMid arrowSB SymbolMid
 			create-aliased-glyph 'less.lig.dblhead.mid'
 			create-forked-glyph 'less.lig.dblhead.shift1' : glyph-proc
 				Joining.set currentGlyph Joining.Classes.Mid
@@ -171,7 +213,8 @@ glyph-block Symbol-Ligation : begin
 
 		create-glyph 'greater.lig.dblhead' : glyph-proc
 			Joining.set currentGlyph Joining.Classes.Left
-			include : LigationGreaterShape (SymbolMid + dH) (SymbolMid - dH) SB RightSB OperatorStroke OperatorStroke (2 * EqualHalfSpace)
+			# Replaced with an arrow head
+			include : ArrowHead arrowSB SymbolMid arrowRSB SymbolMid
 			create-aliased-glyph 'greater.lig.dblhead.mid'
 			create-forked-glyph 'greater.lig.dblhead.shift1' : glyph-proc
 				Joining.set currentGlyph Joining.Classes.Mid
@@ -180,8 +223,8 @@ glyph-block Symbol-Ligation : begin
 		derive-glyphs 'exclam.lig.htmlcommentstart' null 'exclam' : function [src gr] : glyph-proc
 			Joining.set currentGlyph Joining.Classes.Mid
 			include : refer-glyph src
-			include : Translate (((RightSB - Width) + (Width - arrowExtend / 2)) / 2 - Middle) 0
-			include : HBar.m (-arrowExtend) (RightSB - Width) SymbolMid OperatorStroke
+			# Replaced with a variant that gives the arrow head more space
+			include : HBar.m (-arrowExtend) SB SymbolMid OperatorStroke
 
 		# Right-shifted >
 		define [ArrowShiftTranslate k] : glyph-proc
@@ -191,65 +234,57 @@ glyph-block Symbol-Ligation : begin
 
 		create-glyph 'greater.lig.shift0' : glyph-proc
 			include [refer-glyph 'greater'] AS_BASE ALSO_METRICS
-			create-aliased-glyph 'greater.lig.shift0.head'
-			create-aliased-glyph 'greater.lig.shift0.mid'
-			create-aliased-glyph 'greater.lig.shift0.anti'
-			create-aliased-glyph 'greater.lig.shiftN0'
-			create-aliased-glyph 'greater.lig.shiftN0.mid'
 			create-forked-glyph 'greater.lig.shift0h'      : ArrowShiftTranslate   0.5
 			create-forked-glyph 'greater.lig.shift1'       : ArrowShiftTranslate   1.0
-			create-forked-glyph 'greater.lig.shift1.head'  : ArrowShiftTranslate   1.0
 			create-forked-glyph 'greater.lig.shift1h'      : ArrowShiftTranslate   1.5
 			create-forked-glyph 'greater.lig.shift2'       : ArrowShiftTranslate   2.0
 			create-forked-glyph 'greater.lig.shiftN0h'     : ArrowShiftTranslate (-0.5)
 			create-forked-glyph 'greater.lig.shiftN1'      : ArrowShiftTranslate (-1.0)
-			create-forked-glyph 'greater.lig.shiftN1.anti' : ArrowShiftTranslate (-1.0)
 			create-forked-glyph 'greater.lig.shiftN1h'     : ArrowShiftTranslate (-1.5)
 			create-forked-glyph 'greater.lig.shiftN2'      : ArrowShiftTranslate (-2.0)
 
+		# Use arrow head glyph
+		create-glyph 'greater.lig.shift0.head' : glyph-proc
+			include : ArrowHead arrowSB SymbolMid arrowRSB SymbolMid
+			create-aliased-glyph 'greater.lig.shift0.mid'
+			create-aliased-glyph 'greater.lig.shift0.anti'
+			create-forked-glyph 'greater.lig.shift1.head'  : ArrowShiftTranslate   1.0
+			create-forked-glyph 'greater.lig.shiftN1.anti' : ArrowShiftTranslate (-1.0)
+
 		create-glyph 'less.lig.shift0' : glyph-proc
 			include [refer-glyph 'less'] AS_BASE ALSO_METRICS
-			create-aliased-glyph 'less.lig.shift0.head'
-			create-aliased-glyph 'less.lig.shift0.mid'
-			create-aliased-glyph 'less.lig.shift0.anti'
-			create-aliased-glyph 'less.lig.shiftN0'
-			create-aliased-glyph 'less.lig.shiftN0.mid'
 			create-forked-glyph 'less.lig.shift0h'      : ArrowShiftTranslate (-0.5)
 			create-forked-glyph 'less.lig.shift1'       : ArrowShiftTranslate (-1.0)
-			create-forked-glyph 'less.lig.shift1.head'  : ArrowShiftTranslate (-1.0)
 			create-forked-glyph 'less.lig.shift1h'      : ArrowShiftTranslate (-1.5)
 			create-forked-glyph 'less.lig.shift2'       : ArrowShiftTranslate (-2.0)
 			create-forked-glyph 'less.lig.shiftN0h'     : ArrowShiftTranslate   0.5
 			create-forked-glyph 'less.lig.shiftN1'      : ArrowShiftTranslate   1.0
-			create-forked-glyph 'less.lig.shiftN1.anti' : ArrowShiftTranslate   1.0
 			create-forked-glyph 'less.lig.shiftN1h'     : ArrowShiftTranslate   1.5
 			create-forked-glyph 'less.lig.shiftN2'      : ArrowShiftTranslate   2.0
 
+		# Use arrow head glyph
+		create-glyph 'less.lig.shift0.head' : glyph-proc
+			include : ArrowHead arrowRSB SymbolMid arrowSB SymbolMid
+			create-aliased-glyph 'less.lig.shift0.mid'
+			create-aliased-glyph 'less.lig.shift0.anti'
+			create-forked-glyph 'less.lig.shift1.head'  : ArrowShiftTranslate (-1.0)
+			create-forked-glyph 'less.lig.shiftN1.anti' : ArrowShiftTranslate   1.0
+
 		define [ArrowHeadHole] : Rect (SymbolMid + EqualHalfSpace - 0.5 * OperatorStroke) (SymbolMid - EqualHalfSpace + 0.5 * OperatorStroke) (-Width) (2 * Width)
 
 		create-glyph 'less.lig.hole' : glyph-proc
 			Joining.set currentGlyph Joining.Classes.Left
 			include : difference
-				glyph-proc : include : refer-glyph "less"
+				glyph-proc : include : refer-glyph "less.lig.shift0.head"
 				ArrowHeadHole
+			create-forked-glyph 'less.lig.hole.shiftN1' : ArrowShiftTranslate 1.0
 
 		create-glyph 'greater.lig.hole' : glyph-proc
 			Joining.set currentGlyph Joining.Classes.Right
 			include : difference
-				glyph-proc : include : refer-glyph "greater"
-				ArrowHeadHole
-
-		create-glyph 'less.lig.hole.shiftN1' : glyph-proc
-			Joining.set currentGlyph Joining.Classes.Mid
-			include : difference
-				glyph-proc : include : refer-glyph "less.lig.shiftN1"
-				ArrowHeadHole
-
-		create-glyph 'greater.lig.hole.shiftN1' : glyph-proc
-			Joining.set currentGlyph Joining.Classes.Mid
-			include : difference
-				glyph-proc : include : refer-glyph "greater.lig.shiftN1"
+				glyph-proc : include : refer-glyph "greater.lig.shift0.head"
 				ArrowHeadHole
+			create-forked-glyph 'greater.lig.hole.shiftN1' : ArrowShiftTranslate (-1.0)
 
 	do "bottom sign derivatives"
 		glyph-block-import Symbol-Punctuation-Dashes : UnderScoreConfig
@@ -260,12 +295,15 @@ glyph-block Symbol-Ligation : begin
 				include : HBar.b (O - SB) RightSB yBar
 			create-glyph "underscore.lig.fm.\(suffix)" : glyph-proc
 				Joining.set currentGlyph Joining.Classes.Right
-				include : HBar.b SB (Width + RodBarHOffset) yBar
+				# Shift instead of grow
+				include : HBar.b (SB + RodBarHOffset) (Width + RodBarHOffset) yBar
 			create-glyph "underscore.lig.mf.\(suffix)" : glyph-proc
 				Joining.set currentGlyph Joining.Classes.Left
-				include : HBar.b (0 - RodBarHOffset) RightSB yBar
+				# Shift instead of grow
+				include : HBar.b (0 - RodBarHOffset) (RightSB - RodBarHOffset) yBar
 			create-glyph "bar.lig.bottomMid.\(suffix)"
-				VBar.m Middle yBar ParenTop
+				# Middle-align to symbols the bar for logic symbols
+				VBar.m Middle yBar (SymbolMid * 2  - yBar)
 
 		select-variant "underscore.lig.zf" (follow -- 'underscore')
 		select-variant "underscore.lig.fm" (follow -- 'underscore')
@@ -446,3 +484,74 @@ glyph-block Symbol-Ligation : begin
 		create-glyph 'greaterArrow' : composite-proc
 			refer-glyph 'hyphen.lig.jp'
 			refer-glyph 'greater'
+
+	do 'Zig-zag carets'
+		create-glyph 'asciiCaret.lig.zigzag' : glyph-proc
+			local a   : Math.PI / 5
+			local t     aboveMarkTop
+			local b   : t - Width / 2 * [Math.tan a]
+			local swV : OperatorStroke / [Math.cos a]
+			include : spiro-outline
+				corner 0           (b - swV)
+				corner 0           b
+				corner (Width / 2) t
+				corner Width       b
+				corner Width       (b - swV)
+				corner (Width / 2) (t - swV)
+
+	do 'Zig-zag slashes'
+		define [ZigZagShape l r] : glyph-proc
+			local a    : Math.PI / 4
+			local t    : SymbolMid + Width / 2 * [Math.tan a]
+			local b    : SymbolMid - Width / 2 * [Math.tan a]
+			local hswV : Stroke / (2 * [Math.cos a])
+			include : spiro-outline
+				corner r (t - hswV)
+				corner r (t + hswV)
+				corner l (b + hswV)
+				corner l (b - hswV)
+
+		create-glyph 'slash.lig.zig' : glyph-proc
+			include : ZigZagShape 0 Width
+
+		create-glyph 'backslash.lig.zag' : glyph-proc
+			include : ZigZagShape Width 0
+
+	do 'Escape Sequence'
+		local escapeSeqKern : 1 * OperatorStroke
+		create-glyph 'backslash.lig.escape' : glyph-proc
+			Joining.set currentGlyph Joining.Classes.Right
+			include : refer-glyph 'backslash'
+			include : Translate escapeSeqKern 0
+
+	do 'Logical /\ \/'
+		local LogicStroke   OperatorStroke
+		local LogicWid    : XH * [Math.tan (Math.PI / 6)]
+
+		define [LogicAndOrShape l r t b] : glyph-proc
+			local corA : Math.atan ((t - b) / (r - l))
+			local corV : 1/2 * (1 / [Math.cos corA])
+			local corH : 1/2 * (1 / [Math.sin corA])
+			include : spiro-outline
+				corner r (t - LogicStroke * corV)
+				corner r (t + LogicStroke * corV)
+				corner (l - LogicStroke * corH) b
+				corner (l + LogicStroke * corH) b
+
+		# And /\
+		create-glyph 'slash.left' : glyph-proc
+			Joining.set currentGlyph Joining.Classes.Left
+			include : LogicAndOrShape (Width - LogicWid) Width XH 0
+
+		create-glyph 'backslash.right' : glyph-proc
+			Joining.set currentGlyph Joining.Classes.Right
+			include : LogicAndOrShape LogicWid 0 XH 0
+
+		# Or \/
+		create-glyph 'backslash.left' : glyph-proc
+			Joining.set currentGlyph Joining.Classes.Left
+			include : LogicAndOrShape (Width - LogicWid) Width 0 XH
+
+		create-glyph 'slash.right' : glyph-proc
+			Joining.set currentGlyph Joining.Classes.Right
+			include : LogicAndOrShape LogicWid 0 0 XH
diff --git a/packages/font-glyphs/src/symbol/math/arith.ptl b/packages/font-glyphs/src/symbol/math/arith.ptl
index e5ca31804..9bf5d219f 100644
--- a/packages/font-glyphs/src/symbol/math/arith.ptl
+++ b/packages/font-glyphs/src/symbol/math/arith.ptl
@@ -80,7 +80,9 @@ glyph-block Symbol-Math-Arith : begin
 			include : VBar.m ([mix SB RightSB (3/4)] + (1/8) * [HSwToV OperatorStroke]) PlusTop PlusBot fine
 
 	do "Multiply"
-		define Size : 0.5 * (RightSB - SB) + 0.25 * SB
+		# Make multiply the same visual size as a plus operator
+		# TODO: Does this break bowtie glyphs?
+		define Size : 0.5 * (RightSB - SB) - 0.5 * OperatorStroke
 		define swBowtie : AdviceStroke 4.25
 		define swBowtieHv : AdviceStroke 4.75
 
@@ -117,6 +119,10 @@ glyph-block Symbol-Math-Arith : begin
 			include : refer-glyph "multiply"
 			include : HBar.b (Middle - Size) (Middle + Size) 0 OperatorStroke
 
+		# Used for the x multiply ligature
+		create-glyph "x.multiply" : glyph-proc
+			include : refer-glyph "multiply"
+
 		alias 'vectorOrCrossProduct' 0x2A2F 'multiply'
 
 		create-glyph 'multiplyInMulDiv' : glyph-proc
diff --git a/packages/font-glyphs/src/symbol/punctuation/ascii-marks.ptl b/packages/font-glyphs/src/symbol/punctuation/ascii-marks.ptl
index 9ece5e7c9..059a66e95 100644
--- a/packages/font-glyphs/src/symbol/punctuation/ascii-marks.ptl
+++ b/packages/font-glyphs/src/symbol/punctuation/ascii-marks.ptl
@@ -21,7 +21,10 @@ glyph-block Symbol-Punctuation-Ascii-Marks : begin
 
 	create-glyph 'asciiTilde.low' : composite-proc
 		refer-glyph 'asciiTilde.high'
-		ApparentTranslate 0 (SymbolMid - XH - AccentStackOffset)
+		# Adjust tilde to be center-aligned
+		# TODO: I'm sure there's a more correct formula to represent this, however
+		# given current metrics this looks correct.
+		ApparentTranslate 0 (SymbolMid - XH - (AccentStackOffset * 0.75))
 		MarkSet.plus
 	create-glyph 'asciiCaret.medium' : composite-proc
 		refer-glyph 'asciiCaret.high'
diff --git a/packages/font-glyphs/src/symbol/punctuation/ligation-variants.ptl b/packages/font-glyphs/src/symbol/punctuation/ligation-variants.ptl
index 6672eb49a..9e1c477ea 100644
--- a/packages/font-glyphs/src/symbol/punctuation/ligation-variants.ptl
+++ b/packages/font-glyphs/src/symbol/punctuation/ligation-variants.ptl
@@ -26,6 +26,8 @@ glyph-block Symbol-Punctuation-Ligation-Variants : begin
 		* "question"
 
 	define deltas : object
+		# Used in spread op
+		"dRight2"      2.0
 		"dRight"       1.0
 		"dRightHalf"   0.5
 		"dMid"         0
@@ -33,7 +35,8 @@ glyph-block Symbol-Punctuation-Ligation-Variants : begin
 		"dLeft"      (-1.0)
 
 	define [LigD baseName suffix pDelta] : if [glyph-is-needed baseName] : begin
-		local delta : Math.max 0 : Width / 2 - (XH - PeriodSize) / 2
+		# More aggressive dot kerning
+		local delta PeriodSize # : Math.max 0 : Width / 2 - (XH - PeriodSize) / 2
 		derive-glyphs (baseName + '.lig.' + suffix) null baseName : function [src gr]
 			composite-proc
 				refer-glyph src
diff --git a/packages/font-glyphs/src/symbol/punctuation/slashes-and-number-sign.ptl b/packages/font-glyphs/src/symbol/punctuation/slashes-and-number-sign.ptl
index e5545b53b..ea7f203a8 100644
--- a/packages/font-glyphs/src/symbol/punctuation/slashes-and-number-sign.ptl
+++ b/packages/font-glyphs/src/symbol/punctuation/slashes-and-number-sign.ptl
@@ -41,13 +41,14 @@ glyph-block Symbol-Punctuation-Slashes-And-Number-Sign : begin
 		include : SlashShape (slashDefautLeft - b) (slashDefaultRight - b) nothing nothing w
 		include : SlashShape (slashDefautLeft + b) (slashDefaultRight + b) nothing nothing w
 
-	create-glyph 'slash.left' : glyph-proc
-		Joining.set currentGlyph Joining.Classes.Right
-		include : SlashShape slashDefautLeft Width
-
-	create-glyph 'slash.right' : glyph-proc
-		Joining.set currentGlyph Joining.Classes.Left
-		include : SlashShape 0 slashDefaultRight
+# Removed for those defined in ligation.ptl
+#	create-glyph 'slash.left' : glyph-proc
+#		Joining.set currentGlyph Joining.Classes.Right
+#		include : SlashShape slashDefautLeft Width
+#
+#	create-glyph 'slash.right' : glyph-proc
+#		Joining.set currentGlyph Joining.Classes.Left
+#		include : SlashShape 0 slashDefaultRight
 
 	create-glyph 'slantedParallel' : glyph-proc
 		define w : AdviceStroke 3
@@ -78,13 +79,16 @@ glyph-block Symbol-Punctuation-Slashes-And-Number-Sign : begin
 		include : refer-glyph "backslash"
 		include : HBar.m SB RightSB SymbolMid OverlayStroke
 
-	create-glyph 'backslash.left' : glyph-proc
-		Joining.set currentGlyph Joining.Classes.Right
-		include : BackslashShape (Middle - backslashWidth / 2) Width
-
-	create-glyph 'backslash.right' : glyph-proc
-		Joining.set currentGlyph Joining.Classes.Left
-		include : BackslashShape 0 (Middle + backslashWidth / 2)
+# Removed for those defined in ligation.ptl
+#	create-glyph 'backslash.left' : glyph-proc
+#		Joining.set currentGlyph Joining.Classes.Right
+#		include : SlashShape (Width - LogicWid) Width LogicBot LogicTop
+#		#include : BackslashShape (Middle - backslashWidth / 2) Width
+#
+#	create-glyph 'backslash.right' : glyph-proc
+#		Joining.set currentGlyph Joining.Classes.Left
+#		include : SlashShape 0 LogicWid LogicBot LogicTop
+#		#include : BackslashShape 0 (Middle + backslashWidth / 2)
 
 	do "number sign"
 		define fine : AdviceStroke 2.75
diff --git a/packages/font-glyphs/src/symbol/shared.ptl b/packages/font-glyphs/src/symbol/shared.ptl
index b46689fda..2a6aa628f 100644
--- a/packages/font-glyphs/src/symbol/shared.ptl
+++ b/packages/font-glyphs/src/symbol/shared.ptl
@@ -14,11 +14,16 @@ glyph-block Shared-Symbol-Shapes : begin
 		define WaveMagnitude : dist * (3 / 4) - sw / 2
 		define WaveAdj : TanSlope * WaveMagnitude * (-0.75)
 		define phaseShift : fallback _phaseShift 0
-		define [WaveShapeImpl] : with-params [left right ts te xsJoin xfJoin diagJoinS diagJoinF] : glyph-proc
+		# Adds shapeFreq, which uses a power function to increase wave frequency from a given end
+		define [WaveShapeImpl] : with-params [left right ts te xsJoin xfJoin diagJoinS diagJoinF shapeFreq] : glyph-proc
 			local resolution : Math.ceil (WaveResolution * (te - ts))
 			local knots {}
 			foreach [pr : range 0 till resolution] : begin
-				local theta : (phaseShift + [mix ts te (pr / resolution)]) * Math.PI * 2
+				local rawStep : mix ts te (pr / resolution)
+				local step : if (shapeFreq > 0)
+					[Math.pow (rawStep + 1) shapeFreq] - 1
+					if (shapeFreq < 0) (1 - [Math.pow (2 - rawStep) (-1 * shapeFreq)]) rawStep
+				local theta : (phaseShift + step) * Math.PI * 2
 				local waveRelY : [Math.sin theta] + 0.075 * [Math.sin (theta * 3)]
 				knots.push : g2
 					(WaveAdj * waveRelY) + [mix left right (pr / resolution)]
@@ -33,7 +38,7 @@ glyph-block Shared-Symbol-Shapes : begin
 					list [g2                   xfJoin SymbolMid]
 					list [straight.right.end   xfJoin SymbolMid]
 
-		define [WaveShape] : with-params [left right xsJoin xfJoin diagJoinS diagJoinF [waveCount 1] [unitWidth Width]] : WaveShapeImpl (unitWidth * -left) (unitWidth * (1 + right)) (-left * waveCount) ((1 + right) * waveCount) xsJoin xfJoin diagJoinS diagJoinF
+		define [WaveShape] : with-params [left right xsJoin xfJoin diagJoinS diagJoinF [waveCount 1] [unitWidth Width] [shapeFreq 0]] : WaveShapeImpl (unitWidth * -left) (unitWidth * (1 + right)) (-left * waveCount) ((1 + right) * waveCount) xsJoin xfJoin diagJoinS diagJoinF shapeFreq
 		return WaveShape
 
 	glyph-block-export TriangleShape
diff --git a/packages/font-otl/src/gsub-ligation.ptl b/packages/font-otl/src/gsub-ligation.ptl
index f3301b5d5..bda0faade 100644
--- a/packages/font-otl/src/gsub-ligation.ptl
+++ b/packages/font-otl/src/gsub-ligation.ptl
@@ -11,14 +11,14 @@ define look-around null
 define advance : lambda [t] null
 define ident : lambda [t] : t.map : lambda [x] x
 
-export : define [buildLigations gsub para plm] : begin
+export : define [buildLigations gsub glyphStore para plm] : begin
 	# Initialize features
 	define features : new Map
 	foreach [ { featureTag } : pairs-of plm] : begin
 		local feature : gsub.addCommonFeature : gsub.createFeature featureTag
 		features.set featureTag feature
 
-	buildLigationsImpl gsub para [DoLigGroupT gsub plm features]
+	buildLigationsImpl gsub glyphStore para [DoLigGroupT gsub plm features]
 
 define [DoLigGroupT gsub plm features] : function [F] : begin
 	define deDupeGroups : new Map
@@ -88,13 +88,14 @@ define-macro LigGroup : syntax-rules
 ###################################################################################################
 ###################################################################################################
 
-define [buildLigationsImpl gsub para $LigGroup$] : begin
+define [buildLigationsImpl gsub glyphStore para $LigGroup$] : begin
 	define { chain-rule reverse-rule } : gsub.ChainRuleBuilder
 
 	define less {'less'}
-	define lessAndEquiv {'less' 'less.lig.shift0' 'less.lig.shift0.anti'}
+	# TODO: are these equiv the right ones? need to investigate
+	define lessAndEquiv {'less' 'less.lig.shift0' } #'less.lig.shift0.anti'}
 	define greater {'greater'}
-	define greaterAndEquiv {'greater' 'greater.lig.shift0' 'greater.lig.shift0.anti'}
+	define greaterAndEquiv {'greater' 'greater.lig.shift0' } # 'greater.lig.shift0.anti'}
 	define lessOrGreater         : {}.concat less         greater
 	define lessOrGreaterAndEquiv : {}.concat lessAndEquiv greaterAndEquiv
 
@@ -105,12 +106,98 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 
 	define bar {'bar'}
 	define slash {'slash'}
+	define period {'period'}
 	define exclam {'exclam'}
 	define backslash {'backslash'}
 	define underscore {'underscore'}
 	define anyBar {'bar' 'slash' 'backslash'}
 
-	define regexLookAround [less.concat hyphen equal exclam greater anyBar]
+	define regexLookAround [less.concat hyphen equal exclam greater anyBar period]
+
+	# Adds ligatures for x multiply and numeric literals
+	define alpha {}
+	foreach [j : range 0 26] : alpha.push
+		glyphStore.queryNameByUnicode (['a'.charCodeAt 0] + j)
+		glyphStore.queryNameByUnicode (['A'.charCodeAt 0] + j)
+
+	define digit {}
+	foreach [j : range 0 9] : digit.push
+		glyphStore.queryNameByUnicode (['0'.charCodeAt 0] + j)
+
+	define zero       {digit.0}
+	define binDigit : digit.slice 0 2
+	define octDigit : digit.slice 0 8
+	define hexDigit : digit.concat : alpha.slice 0 12
+	define numeric  : digit.concat period underscore
+	define alphaNum : alpha.concat numeric
+
+	define numericLiterals {{{'b'} binDigit} {{'o'} octDigit} {{'x'} hexDigit}}
+
+	LigGroup "Numeric literals" : if [hasLG 'numeric-literals'] : begin
+		foreach { baseLetter baseDigit } numericLiterals : begin
+			local notBaseDigit : alphaNum.filter : x => : not : baseDigit.includes x
+
+			CreateLigationLookup : list
+				chain-rule
+					alphaNum            ~> look-around
+					zero                ~> look-around
+					baseLetter          ~> advance
+				chain-rule
+					baseLetter          ~> advance
+					baseDigit           ~> look-around
+					notBaseDigit        ~> look-around
+				chain-rule
+					baseLetter          ~> advance
+					baseDigit           ~> look-around
+					baseDigit           ~> look-around
+					notBaseDigit        ~> look-around
+				chain-rule
+					baseLetter          ~> advance
+					baseDigit           ~> look-around
+					baseDigit           ~> look-around
+					baseDigit           ~> look-around
+					notBaseDigit        ~> look-around
+				chain-rule
+					zero                ~> look-around
+					baseLetter          ~> {"numericbase{\(baseLetter.0)}"}
+					baseDigit           ~> look-around
+
+	LigGroup "X-Multiply" : if [hasLG 'x-multiply'] : begin
+		CreateLigationLookup : list
+			chain-rule
+				alpha          ~> look-around
+				numeric        ~> look-around
+				numeric        ~> look-around
+				digit          ~> look-around
+				{'x'}          ~> advance
+			chain-rule
+				alpha          ~> look-around
+				numeric        ~> look-around
+				digit          ~> look-around
+				{'x'}          ~> advance
+			chain-rule
+				alpha          ~> look-around
+				digit          ~> look-around
+				{'x'}          ~> advance
+			chain-rule
+				{'x'}          ~> advance
+				digit          ~> look-around
+				alpha          ~> look-around
+			chain-rule
+				{'x'}          ~> advance
+				digit          ~> look-around
+				numeric        ~> look-around
+				alpha          ~> look-around
+			chain-rule
+				{'x'}          ~> advance
+				digit          ~> look-around
+				numeric        ~> look-around
+				numeric        ~> look-around
+				alpha          ~> look-around
+			chain-rule # 2x2
+				digit          ~> look-around
+				{'x'}          ~> {'x.multiply'}
+				digit          ~> look-around
 
 	LigGroup "Bracket-star" : if [hasLG 'brst'] : begin
 		CreateLigationLookup : list
@@ -171,8 +258,17 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 		define opcTo   : opcGroups.map : x => x.right.0
 		define C       : opcFrom ~> opcTo
 
+		# Do not center operators within regex groups
+		define rP  : {'parenLeft'}   ~> look-around
+		define rQ  : {'question'}    ~> look-around
+		define rLA : regexLookAround ~> advance
+
 		# Triggering
 		CreateLigationLookup : list
+			chain-rule rP rQ rLA rLA
+			chain-rule rP rQ rLA rLA rLA
+			chain-rule rP rQ rLA rLA rLA rLA
+			chain-rule rP rQ rLA rLA rLA rLA rLA
 			chain-rule C C skipL skipL triggerL         # **##+
 			chain-rule C C skipL triggerL               # **#+
 			chain-rule C C triggerL                     # **+
@@ -225,6 +321,24 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 				{'asterisk/sMid/ligComment'} ~> look-around
 				{'asterisk' 'asterisk/slashBot'}  ~> [just 'asterisk/sMid/ligComment']
 
+	# Adds spread operator ligation ...xyz
+	LigGroup "Spread" : if [hasLG 'spread-op'] : begin
+		CreateLigationLookup : list
+			chain-rule
+				period ~> look-around
+				period ~> advance
+				period ~> advance
+				period ~> advance
+			chain-rule
+				period ~> advance
+				period ~> advance
+				period ~> advance
+				period ~> look-around
+			chain-rule
+				period ~> [lsx 'dRight2']
+				period ~> [lsx 'dRight']
+				period ~> [lsx 'dMid']
+
 	LigGroup "Kern Dotty" : if [hasLG 'kern-dotty'] : begin
 		define dottyPunctuationSets : list
 			list { "period" "period/mid" } { "colon" "colon/mid" "question" "exclam" }
@@ -350,9 +464,9 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 				equal      ~> look-around
 				equal      ~> look-around
 			chain-rule # <<<=
-				less       ~> [lsx 'shiftN0h']
+				less       ~> [lsx 'shiftN2']
+				less       ~> [lsx 'shiftN1']
 				less       ~> look-around
-				less       ~> [lsx 'shift0h']
 				equal      ~> look-around
 			chain-rule # =>>>=
 				equal      ~> look-around
@@ -367,9 +481,9 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 				equal      ~> look-around
 				equal      ~> look-around
 			chain-rule # >>>=
-				greater    ~> [lsx 'shift0h']
+				greater    ~> [lsx 'shift2']
+				greater    ~> [lsx 'shift1']
 				greater    ~> look-around
-				greater    ~> [lsx 'shiftN0h']
 				equal      ~> look-around
 			chain-rule # =<<=
 				equal      ~> look-around
@@ -382,8 +496,8 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 				equal      ~> look-around
 				equal      ~> look-around
 			chain-rule # <<=
-				less       ~> [lsx 'shiftN0h']
-				less       ~> [lsx 'shift0h']
+				less       ~> [lsx 'shiftN1']
+				less       ~> look-around
 				equal      ~> look-around
 			chain-rule # =>>=
 				equal      ~> look-around
@@ -396,11 +510,12 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 				equal      ~> look-around
 				equal      ~> look-around
 			chain-rule # >>=
-				greater    ~> [lsx 'shift0h']
-				greater    ~> [lsx 'shiftN0h']
+				greater    ~> [lsx 'shift1']
+				greater    ~> look-around
 				equal      ~> look-around
 
 	LigGroup "HTML Comment" : if [hasLG 'html-comment'] : begin
+		local notBefore : hyphen.concat greater
 		CreateLigationLookup : list
 			chain-rule # <!---
 				less   ~> look-around
@@ -413,6 +528,33 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 				exclam ~> [lsx 'htmlcommentstart']
 				hyphen ~> [lsx 'lxc']
 				hyphen ~> [lsx 'cf']
+			# Add HTML comment ends
+			chain-rule # normal arrow for ---> and <-->
+				notBefore ~> look-around
+				hyphen    ~> advance
+				hyphen    ~> advance
+				greater   ~> advance
+			chain-rule # normal arrow for -->>
+				hyphen  ~> advance
+				hyphen  ~> advance
+				greater ~> advance
+				greater ~> look-around
+			chain-rule # -->
+				hyphen  ~> [lsx 'fc']
+				hyphen  ~> [lsx 'cr']
+				greater ~> look-around
+
+	LigGroup "Escape Sequence" : if [hasLG 'escape-seq'] : begin
+		define escape    : [lsx 'escape'] backslash
+		define escapable : alpha.concat digit backslash
+		CreateLigationLookup : list
+			chain-rule
+				escape    ~> look-around
+				backslash ~> advance
+				escapable ~> look-around
+			chain-rule
+				backslash ~> escape
+				escapable ~> look-around
 
 	LigGroup "Arrows" : begin
 		define singular : hyphen.concat anyWave
@@ -465,16 +607,17 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 				regexLookAround ~> advance
 				regexLookAround ~> advance
 
-			define [CJoinHeadForce]         : if dbl [lsx 'dblhead']        [lsx 'shift0.head']
-			define [CJoinHeadMid]           : if dbl [lsx 'dblhead.mid']    [lsx 'shift0.mid']
-			define [CJoinHeadForceShift1]   : if dbl [lsx 'dblhead.shift1'] [lsx 'shift1.head']
+			# Use simplified double arrow heads
+			define [CJoinHeadForce]           [lsx 'shift0.head']
+			define [CJoinHeadMid]           : if dbl [lsx 'hole']           [lsx 'shift0.mid']
+			define [CJoinHeadForceShift1]     [lsx 'shift1.head']
 			define [CAntiHeadForce]         : if dbl [lsx 'hole']           [lsx 'shift0.anti']
 			define [CAntiHeadForceShiftN1]  : if dbl [lsx 'hole.shiftN1']   [lsx 'shiftN1.anti']
 
 			define [LJoinHeadForce]         : if [hasLG 'arrow-l']                          [CJoinHeadForce]        advance
 			define [LJoinHeadForceLR]       : if ([hasLG 'arrow-l'] || [hasLG 'arrow-lr'])  [CJoinHeadForce]        advance
 			define [LJoinHeadForceShift1]   : if [hasLG 'arrow-l']                          [CJoinHeadForceShift1]  advance
-			define [LJoinHeadForceShift1LR] : if ([hasLG 'arrow-l'] || [hasLG 'arrow-lr'])  [CJoinHeadForce]        advance
+			define [LJoinHeadForceShift1LR] : if ([hasLG 'arrow-l'] || [hasLG 'arrow-lr'])  [CJoinHeadForceShift1]  advance
 			define [LMiddleHead]            : if [hasLG 'counter-arrow-l']                  [CJoinHeadMid] : if [hasLG 'arrow-l'] [LJoinHeadForce] advance
 			define [LAntiHeadForce]         : if [hasLG 'counter-arrow-l']                  [CAntiHeadForce]        advance
 			define [LAntiHeadForceShiftN1]  : if [hasLG 'counter-arrow-l']                  [CAntiHeadForceShiftN1] advance
@@ -482,7 +625,7 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 			define [RJoinHeadForce]         : if [hasLG 'arrow-r']                          [CJoinHeadForce]        advance
 			define [RJoinHeadForceLR]       : if ([hasLG 'arrow-r'] || [hasLG 'arrow-lr'])  [CJoinHeadForce]        advance
 			define [RJoinHeadForceShift1]   : if [hasLG 'arrow-r']                          [CJoinHeadForceShift1]  advance
-			define [RJoinHeadForceShift1LR] : if ([hasLG 'arrow-r'] || [hasLG 'arrow-lr'])  [CJoinHeadForce]        advance
+			define [RJoinHeadForceShift1LR] : if ([hasLG 'arrow-r'] || [hasLG 'arrow-lr'])  [CJoinHeadForceShift1]  advance
 			define [RMiddleHead]            : if [hasLG 'counter-arrow-r']                  [CJoinHeadMid] : if [hasLG 'arrow-r'] [RJoinHeadForce] advance
 			define [RAntiHeadForce]         : if [hasLG 'counter-arrow-r']                  [CAntiHeadForce]        advance
 			define [RAntiHeadForceShiftN1]  : if [hasLG 'counter-arrow-r']                  [CAntiHeadForceShiftN1] advance
@@ -538,13 +681,13 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 
 			begin "Dual / Spaceship"
 				heads.push : chain-rule # < < = > >
-					less    ~> look-around
+					less    ~> [LJoinHeadForceLR]
 					less    ~> [LJoinHeadForceShift1LR]
 					rodCls  ~> look-around
 					greater ~> [RJoinHeadForceShift1LR]
-					greater ~> look-around
+					greater ~> [RJoinHeadForceLR]
 				heads.push : chain-rule # < < = >
-					less    ~> look-around
+					less    ~> [LJoinHeadForceLR]
 					less    ~> [LJoinHeadForceShift1LR]
 					rodCls  ~> look-around
 					greater ~> [RJoinHeadForceLR]
@@ -552,7 +695,8 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 					less    ~> [LJoinHeadForceLR]
 					rodCls  ~> look-around
 					greater ~> [RJoinHeadForceShift1LR]
-					greater ~> look-around
+					greater ~> [RJoinHeadForceLR]
+				# TODO: add in advance rule for actual spaceship operator if lt/gt is disabled
 				heads.push : chain-rule # < = >
 					less    ~> [LJoinHeadForceLR]
 					rodCls  ~> look-around
@@ -562,17 +706,17 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 				heads.push : chain-rule # = > >
 					rodCls  ~> look-around
 					greater ~> [RJoinHeadForceShift1]
-					greater ~> look-around
+					greater ~> [RJoinHeadForceLR]
 				heads.push : chain-rule # = < <
 					rodCls  ~> look-around
 					less    ~> [LAntiHeadForceShiftN1]
-					less    ~> look-around
+					less    ~> [LJoinHeadForceLR]
 				heads.push : chain-rule # > > =
-					greater ~> look-around
+					greater ~> [RJoinHeadForceLR]
 					greater ~> [RAntiHeadForceShiftN1]
 					rodCls  ~> look-around
 				heads.push : chain-rule # < < =
-					less    ~> look-around
+					less    ~> [LJoinHeadForceLR]
 					less    ~> [LJoinHeadForceShift1]
 					rodCls  ~> look-around
 				heads.push : chain-rule # = = >
@@ -695,9 +839,11 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 				chain-rule
 					rodTermLeft          ~> look-around
 					[[CRod 'zf'] rodCls] ~> [[CRod 'zm'] rodCls]
-					anyBar               ~> [lsx 'dMid']
+					# Shrink back bar in arrows
+					anyBar               ~> [lsx 'bottomMid']
 				chain-rule
-					anyBar               ~> [lsx 'dMid']
+					# Shrink back bar in arrows
+					anyBar               ~> [lsx 'bottomMid']
 					[[CRod 'fz'] rodCls] ~> [[CRod 'mz'] rodCls]
 					rodTermRight         ~> look-around
 
@@ -729,13 +875,13 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 				regexLookAround ~> advance
 				regexLookAround ~> advance
 			chain-rule # <<<
-				less       ~> [lsx 'shiftN0h']
+				less       ~> [lsx 'shiftN1']
 				less       ~> look-around
-				less       ~> [lsx 'shift0h']
+				less       ~> [lsx 'shift1']
 			chain-rule # >>>
-				greater    ~> [lsx 'shift0h']
+				greater    ~> [lsx 'shift1']
 				greater    ~> look-around
-				greater    ~> [lsx 'shiftN0h']
+				greater    ~> [lsx 'shiftN1']
 			chain-rule # <<
 				less       ~> [lsx 'shiftN0h']
 				less       ~> [lsx 'shift0h']
@@ -744,10 +890,10 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 				greater    ~> [lsx 'shiftN0h']
 
 		# Cancel alternates of < and > shapes if there are more after them
-		local lAll  { "less"    "less.lig.shift0h"    "less.lig.shiftN0h"    }
-		local lNeut { "less"    "less"                "less"                 }
-		local gAll  { "greater" "greater.lig.shift0h" "greater.lig.shiftN0h" }
-		local gNeut { "greater" "greater"             "greater"              }
+		local lAll  { "less"    "less.lig.shift0h"    "less.lig.shiftN0h"    "less.lig.shift1"    "less.lig.shiftN1"    "less.lig.shift2"    "less.lig.shiftN2"    }
+		local lNeut { "less"    "less"                "less"                 "less"               "less"                "less"               "less"                }
+		local gAll  { "greater" "greater.lig.shift0h" "greater.lig.shiftN0h" "greater.lig.shift1" "greater.lig.shiftN1" "greater.lig.shift2" "greater.lig.shiftN2" }
+		local gNeut { "greater" "greater"             "greater"              "greater"            "greater"             "greater"            "greater"             }
 		local lgAll  : lAll.concat  gAll
 		local lgNeut : lNeut.concat gNeut
 		CreateLigationLookup : list
@@ -780,8 +926,43 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 				lAll ~> lNeut
 				lAll ~> lNeut
 
+	# Zig-zags ^^^^^^
+	LigGroup "zig-zag" : if [hasLG 'zig-zag-caret'] : begin
+		local caret {'asciiCaret'}
+		local zigzag {'asciiCaret.lig.zigzag'}
+		CreateLigationLookup : list
+			chain-rule
+				caret  ~> zigzag
+				caret  ~> zigzag
+				caret  ~> zigzag
+			chain-rule
+				zigzag ~> look-around
+				caret  ~> zigzag
+
+	# Zig-zags /\/\/\/
+	LigGroup "zig-zag" : if [hasLG 'zig-zag-slash'] : begin
+		local zig {'slash.lig.zig'}
+		local zag {'backslash.lig.zag'}
+		CreateLigationLookup : list
+			chain-rule
+				slash     ~> zig
+				backslash ~> zag
+				slash     ~> zig
+			chain-rule
+				backslash ~> zag
+				slash     ~> zig
+				backslash ~> zag
+			chain-rule
+				zig       ~> look-around
+				backslash ~> zag
+			chain-rule
+				zag       ~> look-around
+				slash     ~> zig
+
 	# /\ and \/
+	# Updated some of the shapes within
 	LigGroup "logical" : if [hasLG 'logic'] : begin
+		local anyRodOrBar : anyStraightRod.concat bar
 		CreateLigationLookup : list
 			chain-rule
 				slash          ~> [csx 'left']
@@ -790,11 +971,22 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 				backslash      ~> [csx 'left']
 				slash          ~> [csx 'right']
 			chain-rule
-				bar            ~> [lsx "dMid"]
-				anyStraightRod ~> [lsx 'mf']
+				anyRodOrBar    ~> look-around
+				anyRodOrBar    ~> advance
+				anyRodOrBar    ~> advance
 			chain-rule
-				anyStraightRod ~> [lsx 'fm']
-				bar            ~> [lsx "dMid"]
+				anyRodOrBar    ~> advance
+				anyRodOrBar    ~> advance
+				anyRodOrBar    ~> look-around
+			# |-
+			chain-rule
+				bar            ~> [lsx "bottomMid"]
+				anyStraightRod ~> [lsx 'mf.withoutNotch']
+			# -|
+			chain-rule
+				anyStraightRod ~> [lsx 'fm.withoutNotch']
+				bar            ~> [lsx "bottomMid"]
+			# _|_
 			chain-rule
 				underscore     ~> [lsx 'fm']
 				bar            ~> [lsx 'bottomMid']
@@ -1018,12 +1210,14 @@ define [buildLigationsImpl gsub para $LigGroup$] : begin
 					[src] ~> [sfx "cc"]
 					[src] ~> [sfx "cf"]
 				chain-rule # Start 3
-					[src] ~> [sfx "fc"]
+				    # Modified to use variable frequency ends
+					[src] ~> [sfx "f1c"]
 					[src] ~> [sfx "cc"]
-					[src] ~> [sfx "cf"]
+					[src] ~> [sfx "cf1"]
 				chain-rule # Start 2
-					[src] ~> [if [hasLG "\(gr)-\(gr)"] [sfx "fc"] advance]
-					[src] ~> [if [hasLG "\(gr)-\(gr)"] [sfx "cf"] advance]
+				    # Modified to use variable frequency ends
+					[src] ~> [if [hasLG "\(gr)-\(gr)"] [sfx "f1c"] advance]
+					[src] ~> [if [hasLG "\(gr)-\(gr)"] [sfx "cf1"] advance]
 			CreateLigationLookup : list
 				chain-rule
 					{[sfx "jf"]} ~> {[sfx "jc"]}
diff --git a/packages/font-otl/src/gsub-texture.ptl b/packages/font-otl/src/gsub-texture.ptl
index f5286c1bd..c74356315 100644
--- a/packages/font-otl/src/gsub-texture.ptl
+++ b/packages/font-otl/src/gsub-texture.ptl
@@ -9,7 +9,8 @@ export : define [buildGsubTexture gsub glyphStore markGlyphs cvs] : begin
 	local anyMark : Array.from markGlyphs.all
 	local rec : gsub.beginBlock
 
-	define txtr : gsub.addCommonFeature : gsub.createFeature 'TXTR'
+	# Renames TXTR to kern to enable by default in most environments
+	define txtr : gsub.addCommonFeature : gsub.createFeature 'kern'
 	define { chain-rule reverse-rule } : gsub.ChainRuleBuilder
 	define look-around null
 
diff --git a/packages/font-otl/src/index.ptl b/packages/font-otl/src/index.ptl
index bb5e89048..e609274e5 100644
--- a/packages/font-otl/src/index.ptl
+++ b/packages/font-otl/src/index.ptl
@@ -53,7 +53,7 @@ define [buildGSUB para glyphStore markGlyphs] : begin
 		define plm : Object.assign {.} para.ligation.defaultBuildup
 		if (para.ligation.caltBuildup && para.ligation.caltBuildup.length) : begin
 			set plm.calt para.ligation.caltBuildup
-		buildLigations gsub para plm
+		buildLigations gsub glyphStore para plm
 
 	# frac
 	buildFrac gsub glyphStore
diff --git a/packages/param/src/metric-override.mjs b/packages/param/src/metric-override.mjs
index 99f7bc134..1a84f0c38 100644
--- a/packages/param/src/metric-override.mjs
+++ b/packages/param/src/metric-override.mjs
@@ -4,7 +4,7 @@ export function applyMetricOverride(para, mo, argv) {
 	const bindings = initBindings(para, argv);
 	for (const [field, expr] of Object.entries(mo)) {
 		if (!validMetricOverrideFields.has(field)) {
-			console.error(`Field ${field} cannot be get overridden.`);
+			console.error(`Field ${field} cannot be get overridden.`, expr);
 			continue;
 		}
 		if (typeof expr === "number") {
diff --git a/params/ligation-set.toml b/params/ligation-set.toml
index 77834a1d9..c217b06c8 100644
--- a/params/ligation-set.toml
+++ b/params/ligation-set.toml
@@ -191,8 +191,8 @@ sampleRank = 2
 desc = 'Enable ligation for `<<=`, `>>=` as shift operator'
 
 [simple.html-comment]
-samples = ["<!--", "<!---"]
-desc = 'Enable ligation for `<!--` and `<!---`'
+samples = ["<!--", "<!---", "-->"]
+desc = 'Enable ligation for `<!--`, `<!---`, `-->` and `--->`'
 
 [simple.colon-greater-as-colon-arrow]
 samples = [":>"]
@@ -253,6 +253,30 @@ desc = 'Make 2 or more contiguous hash signs (number signs) (like `##`, `###` an
 samples = ["###", "####"]
 desc = 'Make 3 or more contiguous hash signs (number signs) (like `##` and `###`) connected'
 
+[simple.x-multiply]
+samples = ["2x2"]
+desc = 'Display the letter x as a multiply when between two numbers'
+
+[simple.numeric-literals]
+samples = ["0xF00D", "0o7104", "0b10010111"]
+desc = 'Shows the base in subscript within a numeric literal'
+
+[simple.zig-zag-caret]
+samples = ["^^^", "^^^^"]
+desc = 'Make 3 or more contiguous caret (like `^^^`) a connected zig-zag pattern'
+
+[simple.zig-zag-slash]
+samples = ["/\\/\\/", "\\/\\/\\/\\"]
+desc = 'Shows alternating slash and backslash in a continuous zig-zag pattern'
+
+[simple.escape-seq]
+samples = ["\\\\\\\\", "\\0", "line end\\n"]
+desc = 'Kerns escape sequence backslash to the character being escaped'
+
+[simple.spread-op]
+samples = ["...xyz"]
+desc = 'Kerns three dots right as a spread operator'
+
 ###################################################################################################
 
 # Non-tagged composites, used for inheritance
diff --git a/params/parameters.toml b/params/parameters.toml
index e06f0199e..808c754d7 100644
--- a/params/parameters.toml
+++ b/params/parameters.toml
@@ -35,9 +35,9 @@ accentWidth = 200       # Width of accent marks.
 accentClearance = 80    # Vertical clearance of accent marks to the base.
 accentHeight = 140      # Height of accent marks.
 accentStackOffset = 205 # Offset height of accent mark stack.
-overshoot = -8          # Vertical overshoot for arcs.
-overshootx = -8         # Horizontal overshoot for arcs.
-oxhook = -2             # Horizontal overshoot for hook tips.
+overshoot = -12         # Vertical overshoot for arcs.
+overshootx = -12        # Horizontal overshoot for arcs.
+oxhook = -4             # Horizontal overshoot for hook tips.
 
 refJut = 85
 
