diff --git a/packages/font-glyphs/src/auto-build/transformed.ptl b/packages/font-glyphs/src/auto-build/transformed.ptl
index 39e7da7..e9ab0a3 100644
--- a/packages/font-glyphs/src/auto-build/transformed.ptl
+++ b/packages/font-glyphs/src/auto-build/transformed.ptl
@@ -731,6 +731,14 @@ glyph-block Autobuild-Transformed-Texture : begin
 		# link-relations relSets
 
 	define ranges : list
+		list 0x20 0x21
+		list 0x27 0x29
+		list 0x2C 0x2C
+		list 0x2E 0x2E
+		list 0x2E 0x2E
+		list 0x5B 0x5D
+		list 0x60 0x60
+		list 0x6B 0x6D
 		list 0x41 0x5A
 		list 0x61 0x7A
 		list 0xC0 0xFF
@@ -747,8 +755,8 @@ glyph-block Autobuild-Transformed-Texture : begin
 				if source : results.push { null source }
 		return results
 
-	define EXTENSION :  1 / 12
-	define SHRINK    : -1 / 12
+	define EXTENSION :  1 / 8
+	define SHRINK    : -1 / 8
 	createTextureDerivatives Texture.ExtL  EXTENSION 0         [jobs 0xF000]
 	createTextureDerivatives Texture.ExtR  0         EXTENSION [jobs 0xF100]
 	createTextureDerivatives Texture.ExtLR EXTENSION EXTENSION [jobs 0xF200]
diff --git a/packages/font-glyphs/src/symbol/ligation.ptl b/packages/font-glyphs/src/symbol/ligation.ptl
index 5406718..1e401f5 100644
--- a/packages/font-glyphs/src/symbol/ligation.ptl
+++ b/packages/font-glyphs/src/symbol/ligation.ptl
@@ -26,7 +26,7 @@ glyph-block Symbol-Ligation : begin
 	define RodBarHOffset : 0.5 * Width - [HSwToV HalfStroke] - RodBarHGap
 
 	# Arrow parameters
-	define arrowheadSlope : 1 / LessSlope
+	define arrowheadSlope : 1 / 2 # LessSlope
 
 	local arrowExtend : clamp 0 Width (RightSB - 0.5 * OperatorStroke * arrowheadSlope)
 	local arrowJ      : clamp 0 Width (SB + 0.5 * OperatorStroke * arrowheadSlope)
@@ -180,7 +180,7 @@ glyph-block Symbol-Ligation : begin
 		derive-glyphs 'exclam.lig.htmlcommentstart' null 'exclam' : function [src gr] : glyph-proc
 			Joining.set currentGlyph Joining.Classes.Mid
 			include : refer-glyph src
-			include : Translate (((RightSB - Width) + (Width - arrowExtend / 2)) / 2 - Middle) 0
+			# include : Translate (((RightSB - Width) + (Width - arrowExtend / 2)) / 2 - Middle) 0
 			include : HBar.m (-arrowExtend) (RightSB - Width) SymbolMid OperatorStroke
 
 		# Right-shifted >
diff --git a/packages/font-otl/src/gsub-texture.ptl b/packages/font-otl/src/gsub-texture.ptl
index f5286c1..a77d5f5 100644
--- a/packages/font-otl/src/gsub-texture.ptl
+++ b/packages/font-otl/src/gsub-texture.ptl
@@ -9,7 +9,7 @@ export : define [buildGsubTexture gsub glyphStore markGlyphs cvs] : begin
 	local anyMark : Array.from markGlyphs.all
 	local rec : gsub.beginBlock
 
-	define txtr : gsub.addCommonFeature : gsub.createFeature 'TXTR'
+	define txtr : gsub.addCommonFeature : gsub.createFeature 'kern'
 	define { chain-rule reverse-rule } : gsub.ChainRuleBuilder
 	define look-around null
 
