diff --git a/packages/font-glyphs/src/symbol/ligation.ptl b/packages/font-glyphs/src/symbol/ligation.ptl
index 5406718..1e401f5 100644
--- a/packages/font-glyphs/src/symbol/ligation.ptl
+++ b/packages/font-glyphs/src/symbol/ligation.ptl
@@ -26,7 +26,7 @@ glyph-block Symbol-Ligation : begin
 	define RodBarHOffset : 0.5 * Width - [HSwToV HalfStroke] - RodBarHGap
 
 	# Arrow parameters
-	define arrowheadSlope : 1 / LessSlope
+	define arrowheadSlope : 1 / 2 # LessSlope
 
 	local arrowExtend : clamp 0 Width (RightSB - 0.5 * OperatorStroke * arrowheadSlope)
 	local arrowJ      : clamp 0 Width (SB + 0.5 * OperatorStroke * arrowheadSlope)
@@ -180,7 +180,7 @@ glyph-block Symbol-Ligation : begin
 		derive-glyphs 'exclam.lig.htmlcommentstart' null 'exclam' : function [src gr] : glyph-proc
 			Joining.set currentGlyph Joining.Classes.Mid
 			include : refer-glyph src
-			include : Translate (((RightSB - Width) + (Width - arrowExtend / 2)) / 2 - Middle) 0
+			# include : Translate (((RightSB - Width) + (Width - arrowExtend / 2)) / 2 - Middle) 0
 			include : HBar.m (-arrowExtend) (RightSB - Width) SymbolMid OperatorStroke
 
 		# Right-shifted >
